beesdoo_shift_swap/models/dated_template.py

```diff
@@ -0,0 +1,175 @@
+from odoo import models, fields, api
+from datetime import datetime, timedelta
+
+
+class DatedTemplate(models.Model):
+    _name = 'beesdoo.shift.template.dated'
+
+    date = fields.Datetime(required = True)
+    template_id = fields.Many2one("beesdoo.shift.template")
+    store = fields.Boolean(string="store", invisible=True)
+    hour = fields.Integer(string='Hour',compute='_compute_time', store=True)
+
+    @api.depends('date')
+    def _compute_time(self):
+        for record in self:
+            if not record.date :
+                record.hour = False
+            else :
+                record.hour = int(record.date.strftime('%H:%M:%S').replace(":", ""))
+
+    @api.multi
+    def name_get(self):
```

Comment:
I'm not sure if the name `name_get` is mandatory, but beggining a function name with a verb is a good standard. So `get_name` will be nicer.